File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Config/Db4oLegacyConfigurationBridge.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Config;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Config;

namespace Db4objects.Db4o.Internal.Config
{
	public class Db4oLegacyConfigurationBridge
	{
		public static IEmbeddedConfiguration AsEmbeddedConfiguration(IConfiguration legacy
			)
		{
			return new EmbeddedConfigurationImpl(legacy);
		}

		public static ICommonConfiguration AsCommonConfiguration(IConfiguration config)
		{
			return new CommonConfigurationImpl((Config4Impl)config);
		}

		public static Config4Impl AsLegacy(object config)
		{
			return ((ILegacyConfigurationProvider)config).Legacy();
		}

		public static IFileConfiguration AsFileConfiguration(IConfiguration config)
		{
			return new FileConfigurationImpl((Config4Impl)config);
		}

		public static IIdSystemConfiguration AsIdSystemConfiguration(IConfiguration config
			)
		{
			return new IdSystemConfigurationImpl((Config4Impl)config);
		}
	}
}

File: db4oj/core/src/com/db4o/internal/config/Db4oLegacyConfigurationBridge.java
/* Copyright (C) 2009  Versant Inc.   http://www.db4o.com */
package com.db4o.internal.config;

import com.db4o.config.*;
import com.db4o.internal.*;

public class Db4oLegacyConfigurationBridge {

	public static EmbeddedConfiguration asEmbeddedConfiguration(Configuration legacy) {
		return new EmbeddedConfigurationImpl(legacy);
	}

	public static CommonConfiguration asCommonConfiguration(Configuration config) {
		return new CommonConfigurationImpl((Config4Impl) config);
	}

	public static Config4Impl asLegacy(final Object config) {
		return ((LegacyConfigurationProvider)config).legacy();
	}

	public static FileConfiguration asFileConfiguration(Configuration config) {
		return new FileConfigurationImpl((Config4Impl)config);
	}
	
	public static IdSystemConfiguration asIdSystemConfiguration(Configuration config){
		return new IdSystemConfigurationImpl((Config4Impl) config);
	}

}
